### ISSUE TITLE
Add an exercise about confidence in changes

### ISSUE THREAD

**Issue by chillenzer** [edbennett/python-testing-ci#19]
One important advantage of good code coverage and an extensive test suite is confidence when it comes to code changes. Maybe it would be nice to include an exercise about this, although I'm afraid that this would be a little extensive. I'm thinking about something along the lines of

* take an example code (potentially the `grid` one) that is sufficiently complex (and/or lacking SRP)
* propose a change that will affect several aspects (and assume that the learners won't spot all of them in a small given amount of time)
* ask for a list of changes needed (not necessarily implement them for the sake of saving time) and their confidence in doing so
* then reveal the full list and (in the hope that they missed something) discuss how an exhaustive test suite would have caught their oversights

**Comment by edbennett** [comment]
I think that would be a very tricky exercise to write, but would be valuable if it could be written.

**Comment by chillenzer** [comment]
Yes, true. And it has some overlap with the last one that we are also struggling to keep alive. Maybe it would suffice to point said advantage out more clearly in that exercise.
